icommand.lisp

```diff
@@ -38,22 +38,22 @@
   (message "~s finished" name)
   (pop-top-map))
 
-(defmacro deficommand (name key-bindings &key pre-start-fn cleanup-fn)
+(defmacro deficommand (name (&key bootstrap-fn cleanup-fn) &body key-bindings)
```

Comment:
Some minor things, As joshe mentioned on IRC, `defcommand` has interactive-args which makes the name confusing. Also having only one different letter in the middle of the word makes it hard to differentiate from `defcommand` when skimming. Other options would be `defhydra`, or `define-interactive-keymap` but I'm not convinced on either of them. @dbjergaard What do you think?  

Another, nitpick, How about the names, on-entry and on-exit instead of bootstrap-fn and cleanup-fn